temp/OnnxTransformTempSolution/Microsoft.ML.Transforms.Onnx/OnnxTransform.cs

```diff
@@ -0,0 +1,294 @@
+                {
+                    if (activeOutput(0))
+                        getters[0] = Utils.MarshalInvoke(MakeGetter<int>, _outputNodeInfo.DataType, input);
+
+                    ch.Done();
+                    return getters;
+                }
+            }
+
+            private Delegate MakeGetter<T>(IRow input)
+            {
+                _host.AssertValue(input);
+                _host.Assert(typeof(T) == _outputColType.ItemType.RawType);
+
+                _idvToTensorAdapter.InitializeValueGetters(input);
+
+                ValueGetter<VBuffer<T>> valuegetter = (ref VBuffer<T> dst) =>
+                {
+                    var outputTensors = _model.Run(new List<Tensor> { _idvToTensorAdapter.GetTensor() });
+                    Contracts.Assert(outputTensors.Count() > 0);
+
+                    var values = dst.Values;
+                    if (Utils.Size(values) < _outputColType.VectorSize)
+                        values = new T[_outputColType.VectorSize];
+
+                    OnnxUtils.CopyTo(outputTensors[0], values);
+                    dst = new VBuffer<T>(values.Length, values);
+                };
+
+                return valuegetter;
+            }
+        }
+
+        /// <summary>
+        /// IdvToTensorAdapter adapts an Idv (row-iterator interface) to a tensor-iterator interface.
+        /// For an Idv, you'd need to create a cursor and iterate over it to get each rows of the Idv.
+        /// After adaptation, you'd call GetTensor() on the IdvToTensorAdapter object to get the Tensor equivalent of 
+        /// each row. 
+        /// </summary>
+        internal sealed class IdvToTensorAdapter
+        {
+            // Idv information
+            private readonly string _idvColumnName;
+            internal readonly int _idvColumnIndex;
+            private readonly bool _idvIsVectorColumn;
+            public readonly ColumnType _idvColumnType;
+
+            // Onnx tensor information
+            private readonly OnnxShape _onnxTensorShape;
+
+            private ITensorValueGetter _tensorValueGetter;
+
+            public IdvToTensorAdapter(ISchema idvSchema, Arguments transformArgs)
+            {
+                _idvColumnName = transformArgs.InputColumn;
+                if (!idvSchema.TryGetColumnIndex(_idvColumnName, out _idvColumnIndex))
+                    throw Contracts.Except($"Column '{_idvColumnName}' does not exist");
+                _idvColumnType = idvSchema.GetColumnType(_idvColumnIndex);
+                _idvIsVectorColumn = _idvColumnType.IsVector;
+                _onnxTensorShape = transformArgs.ModelInfo.InputsInfo[0].Shape;
+
+                // TODO: Check that the idv and tensor sizes match
+                // TODO: Check type matches
+            }
+
+            public void InitializeValueGetters(IRow idvRow)
+            {
+                var type = _idvColumnType.ItemType.RawType;
+                _tensorValueGetter = Utils.MarshalInvoke(
```

Comment:
>_tensorValueGetter [](start = 16, length = 18)

I think it would be better to define a delegate that returns a Tensor,  that way instead of having a mutable object, this method could return it and it can be used in line 184 instead of _idvToTensorAdapter.GetTensor().